src/components/Image/index.js

```diff
@@ -7,67 +7,58 @@ import ONYXKEYS from '../../ONYXKEYS';
 import {defaultProps, imagePropTypes} from './imagePropTypes';
 import RESIZE_MODES from './resizeModes';
 
-class Image extends React.Component {
-    componentDidMount() {
-        this.configureOnLoad();
-    }
-
-    componentDidUpdate(prevProps) {
-        if (prevProps.source === this.props.source) {
-            return;
-        }
-        this.configureOnLoad();
-    }
-
+function Image(props) {
     /**
      * Check if the image source is a URL - if so the `encryptedAuthToken` is appended
      * to the source.
      * @returns {Object} - the configured image source
      */
-    getImageSource() {
-        const source = this.props.source;
+    const getImageSource = useCallback(() => {
+        const source = props.source;
         let imageSource = source;
-        if (this.props.isAuthTokenRequired) {
+        if (props.isAuthTokenRequired) {
             // There is currently a `react-native-web` bug preventing the authToken being passed
             // in the headers of the image request so the authToken is added as a query param.
             // On native the authToken IS passed in the image request headers
-            const authToken = lodashGet(this.props, 'session.encryptedAuthToken', null);
+            const authToken = lodashGet(props, 'session.encryptedAuthToken', null);
             imageSource = {uri: `${source.uri}?encryptedAuthToken=${encodeURIComponent(authToken)}`};
         }
 
         return imageSource;
-    }
+    }, [props]);
```

Comment:
also you should destruct props and use correct deps, value should not be re-calculated based on not used props like session, onLoad and etc